src/onemod/fsutils/path_manager.py

```diff
@@ -0,0 +1,41 @@
+from pathlib import Path
+
+
+class PathManager:
```

Comment:
Several random thoughts here :)
* This class is very similar to the dictionary class, maybe we can directly subclass `dict` to borrow some existing methods.
* One thing I found I frequently doing is to build path on the existing path. For example if we specify the `root` folder, we can create `data` or `config` folder based on the root folder. Might be good to add a method for it

Reply:
I do enjoy your random thoughts :D

- Sure. What methods/use cases are you specifically thinking of?
- Could you elaborate on this? I think I haven't fully understood the idea. How would it be different from using `pathlib` for instance to construct paths?
  - Maybe related - is there any value in having a special `root` directory? I guess this is kind of an idea on an idea. But on second thought as long as our directories management system is intuitive enough it probably doesn't add value.

Reply:
For the dictionary inherent idea. It is more for simplifying the code by subclassing `collections.UserDict`. We only need to create a `add` method, and overwrite `__setitem__` method, and for the `get_path`, `remove_path` we will have the counterpart from dictionary `__getitem__` or `pop` and we can batch update with the `update` as well. I don't have a strong preference on this, just a hutch it might be more future proof.

For the path dependency idea, an concrete example is that we currently have a parent folder called `directory`, and all the `data`, `config` directory is provided relative to the `directory` folder. This allows user to repetitively specify the full path from `directory` and build the path from the existing paths in the object. But as you said, we can just easily explicitly use the path in the object `path_manager.get_path("directory") / "data"` to achieve this as well, and probably more readable and flexible this way. Let me know what you think :)

Reply:
Would the alternative be like an `dataif.root.append_path("data")`? I think it works if there is a special concept of `root`, and maybe there is other future functionality to add around it. Sounds like this is what you are implying, correct? I suppose then we have methods for `get_root()`, `set_root()`.
Or else is it that with `add_path()` we assume all paths are relative to `root` by default and the user needs to specify if they want to provide a non-relative path? This is one of the first implementation approaches I went with. But then I realized there was at least one place already in OneMod where an absolute path (pointing to the Pipeline data file) was specified and after trying out each approach it felt more intuitive (to me, with current OneMod Pipeline, Stages relationship) to assume absolute path by default.
For arbitrary non-special directories, just using `get_path()` probably makes sense.

I'm going to propose another ticket on this one too ^^; Might be a small amount of design discussion still to be had.

Reply:
That sounds good! Thanks very much @blsmxiu47!